modules/vaos/app/models/ccra/referral_detail.rb

```diff
@@ -23,6 +25,32 @@ def initialize(attributes)
       @referral_number = referral['ReferralNumber']
       @phone_number = referral['ProviderPhone'] || referral['FacilityPhone']
       @uuid = nil # Will be set by controller
+
+      # Parse referring facility info
+      parse_referring_facility_info(referral['ReferringFacilityInfo']) if referral['ReferringFacilityInfo'].present?
+    end
+
+    private
+
+    # Parse referring facility info from the CCRA response
+    #
+    # @param facility_info [Hash] The facility info from the CCRA response
+    def parse_referring_facility_info(facility_info)
+      return if facility_info.blank?
+
+      @referring_facility_name = facility_info['FacilityName']
+      @referring_facility_phone = facility_info['Phone']
+      @referring_facility_code = facility_info['FacilityCode']
+
+      # Parse address information
+      if facility_info['Address'].present?
+        @referring_facility_address = {
+          street1: facility_info['Address']['Address1'],
+          city: facility_info['Address']['City'],
+          state: facility_info['Address']['State'],
```

Comment:
Note: I added state even when it is not shown in the API spec. I think that omission is an error and will communicate to the team.